torchbenchmark/models/nvidia_deeprecommender/__init__.py

```diff
@@ -0,0 +1,93 @@
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import RECOMMENDATION
+
+import gc
+from .nvtrain import DeepRecommenderTrainBenchmark
+from .nvinfer import DeepRecommenderInferenceBenchmark
+
+class Model(BenchmarkModel):
+  
+  task = RECOMMENDATION.RECOMMENDATION
+  
+  def __init__(self, device="cpu", jit=False):
+    super().__init__()
+    self.devicename = device
+    self.notimplementedreason = "Implemented"
+    self.evalMode = True #default to inference
+
+    if jit:
+      self.notimplementedreason = "Jit Not Supported"
+
+    elif self.devicename != "cpu" and self.devicename != "cuda":
+      self.notimplementedreason = "device type not supported"
+
+    elif self.devicename == "cuda" and torch.cuda.is_available() == False:
+      self.notimplementedreason = "cuda not available on this device"
+
+    else:
+      self.train_obj = DeepRecommenderTrainBenchmark(device = device, jit = jit)
+      self.infer_obj = DeepRecommenderInferenceBenchmark(device = device, jit = jit)
+
+  def get_module(self):
+    if self.evalMode:
+        return lambda x: self.eval(), [0]
+
+    return lambda x: self.train(), [0]
+
+  def set_eval(self):
+    self.evalMode = True
+    return
+    
+  def set_train(self):
+    self.evalMode = False
+    return
+
+  def train(self, niter=1):
+    self.checkimplemented()
+
+    for i in range(niter):
+      self.train_obj.train(self.train_obj.args.num_epochs)
+  
+    self.cleanup()
```

Comment:
The `cleanup()` call is not needed because it is handled by the framework:  https://github.com/pytorch/benchmark/blob/master/test_bench.py#L48

Reply:
I got memory complaints before I added this. The memory cleanup is called out in the ADDING_MODELS doc. Also isn't the line above is unreached?

Reply:
what were the memory complaints? is it still happening after the other changes you made?  

agreed this shouldn't be necessary, possible there is some underlying bug it's masking